src/h3lib/lib/algos.c

```diff
@@ -155,9 +155,16 @@ static const Direction NEW_ADJUSTMENT_III[7][7] = {
  * Maximum number of cells that result from the gridDisk algorithm with the
  * given k. Formula source and proof: https://oeis.org/A003215
  *
- * @param  k   k value, k >= 0.
+ * @param   k   k value, k >= 0.
+ * @param out   size in indexes
  */
-int H3_EXPORT(maxGridDiskSize)(int k) { return 3 * k * (k + 1) + 1; }
+H3Error H3_EXPORT(maxGridDiskSize)(int k, int64_t *out) {
+    if (k < 0) {
+        return E_DOMAIN;
+    }
+    *out = 3 * (int64_t)k * ((int64_t)k + 1) + 1;
```

Comment:
Would it make sense to check somewhere that this value is less than the max number of cells at the current res? E.g. should `kRing(res0cell, 100)` throw `E_DOMAIN`?

Reply:
I have a concern that in some cases this will overflow, in that case we should consider returning the maximum number of cells?

Reply:
Yeah, that's probably better than returning an error (and should be correct, in that I think k-rings covering more than the total number of cells should return all cells at that res, right?).

We should be able to check a constant to avoid integer overflow, right?